### ISSUE TITLE
Error when installing gem  'sciruby-full'

### ISSUE THREAD

**Issue by vnluc** [oneclick/rubyinstaller2#77]
This project is for Ruby version 2.4 and newer.
For Ruby versions < 2.4 please file an issue [here]( https://github.com/oneclick/rubyinstaller/issues/new ).

## What problems are you experiencing?
Error when installing gem  'sciruby-full'

## Steps to reproduce
$gem install sciruby-full

## What's the output from `ridk version`?

---
ruby:
  path: D:/ruby/ruby_packages_nobk/rubyinstaller-2.5.0-snapshot-x64_aug29
  version: 2.5.0
  platform: x64-mingw32
ruby_installer:
  package_version: 2.5.0-snapshot
  git_commit: 1fd689f
msys2:
  path: D:\msys64
  title: MSYS2 64bit
  version: '20161025'
cc: x86_64-w64-mingw32-gcc (Rev2, Built by MSYS2 project) 7.1.0
sh: GNU bash, version 4.4.12(1)-release (x86_64-pc-msys)
os: Microsoft Windows [Version 6.1.7601]

## error message
$gem install sciruby-full

...
...

Successfully installed gnuplotrb-0.3.4
Fetching: rmagick-2.16.0.gem (100%)
Building native extensions.  This could take a while...
ERROR:  Error installing sciruby-full:
        ERROR: Failed to build gem native extension.

    current directory: D:/ruby/ruby_packages_nobk/rubyinstaller-2.5.0-snapshot-x64_aug29/lib/ruby/ge
ms/2.5.0/gems/rmagick-2.16.0/ext/RMagick
D:/ruby/ruby_packages_nobk/rubyinstaller-2.5.0-snapshot-x64_aug29/bin/ruby.exe -r ./siteconf20170910
-3412-14dm30t.rb extconf.rb
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
        --with-opt-dir
        --without-opt-dir
        --with-opt-include
        --without-opt-include=${opt-dir}/include
        --with-opt-lib
        --without-opt-lib=${opt-dir}/lib
        --with-make-prog
        --without-make-prog
        --srcdir=.
        --curdir
        --ruby=D:/ruby/ruby_packages_nobk/rubyinstaller-2.5.0-snapshot-x64_aug29/bin/$(RUBY_BASE_NAM
E)
extconf.rb:141:in ``': No such file or directory - identify -version (Errno::ENOENT)
        from extconf.rb:141:in `configure_compile_options'
        from extconf.rb:16:in `initialize'
        from extconf.rb:548:in `new'
        from extconf.rb:548:in `<main>'

extconf failed, exit code 1

Gem files will remain installed in D:/ruby/ruby_packages_nobk/rubyinstaller-2.5.0-snapshot-x64_aug29
/lib/ruby/gems/2.5.0/gems/rmagick-2.16.0 for inspection.
Results logged to D:/ruby/ruby_packages_nobk/rubyinstaller-2.5.0-snapshot-x64_aug29/lib/ruby/gems/2.
5.0/extensions/x64-mingw32/2.5.0/rmagick-2.16.0/gem_make.out

**Comment by MSP-Greg** [comment]
Have you looked at the following?
https://github.com/rmagick-temp/rmagick/wiki/Installing-on-Windows

**Comment by vnluc** [comment]
Hi @MSP-Greg, thank you for your information. sciruby depend on many other gems. I will install other gems I need first.
For rmagick I will check, at the moment I not need it yet.

**Comment by MSP-Greg** [comment]
@vnluc 

You're using trunk / 2.5.  Please be aware that every gem you try to install (along with all the dependency gems) will need to be compiled, and to do so, may require additional packages and/or files to be added to the MSYS2/MinGW build system.

Gem builders can add info to the gem spec that allows the RubyInstaller2 runtime to add/install required MSYS2/MinGW packages, but, at present, only the most popular gems have done so.

Simply put, if you've successfully compiled one gem, the MSYS2/MinGW system is probably installed properly and functioning as it should.

So, further issues are probably due to the gems lack of support for windows, and not due to the RubyInstaller2 runtime or MYS2/MinGW.

**Comment by vnluc** [comment]
Hi @MSP-Greg, thank you. 
Yes, I've sucessfully install and compile some gems with native extension (racc, nmatrix, pry...). I think rmagick have not been modified to support on Windows. howerver, I have not been able to check to that step yet.

**Comment by larskanis** [comment]
If RMagick would support the latest ImageMagick version, it could install+use ImageMagick from the [MSYS2/MINGW repository](https://github.com/Alexpux/MINGW-packages/tree/master/mingw-w64-imagemagick) per [gemspec dependency](https://github.com/oneclick/rubyinstaller2/wiki/For-gem-developers#msys2-library-dependency). But [it doesn't](https://github.com/rmagick/rmagick/issues/256).

Fortunately it's cumbersome but possible to install the rmagick gem as [described here](https://medium.com/ruby-on-rails-web-application-development/install-rmagick-gem-on-windows-7-8-10-imagemagick-6-9-4-q16-hdri-5492c3fef202). I already used this procedure several times successfully.

**Comment by vnluc** [comment]
@MSP-Greg , per your explaination above for released Ruby version for Windows. Gems will come pre-compiled? I am not sure of this aspect :)

@larskanis : Thank you, Rmagick is not required for me now but it is great to have information here in case other might need it.

**Comment by MSP-Greg** [comment]
@vnluc 

> @MSP-Greg , per your explanation above for released Ruby version for Windows. Gems will come pre-compiled? I am not sure of this aspect :)

Totally dependent on the Gem builder.  Take Nokogiri as an example.  If you go to the RubyGems.org web site and download `nokogiri-1.8.0-x64-mingw32`, then [`unpack`](https://msp-greg.github.io/ruby_trunk/rubygems/file.command-reference.html#gem-unpack) it, you'll noticed several folders under `lib\nokogiri` like 2.2, 2.3, and 2.4.  This means that the gem is compiled for use with Ruby 2.2 thru 2.4.  One reason this is required is that the Ruby dll's are named version specific.

So, if you want to use 1.8.0, you'll have no choice but to compile it yourself for trunk (2.5) or any version earlier than 2.2.  The normal gem (1.8.0) does support 2.1.

In terms of an easy determination for how much trouble a gem might be to install, if the GitHub repo does not have an appveyor.yml file in the root folder, it probably hasn't been tested with Windows.

**Comment by larskanis** [comment]
Since this is not a RubyInstaller issue, I'm closing this now. See my [comment above](https://github.com/oneclick/rubyinstaller2/issues/77#issuecomment-328439669) for possible solutions to the RMagick issue.
